tests/perf/container_perf.cc

```diff
@@ -233,3 +238,239 @@ PERF_TEST_F(container_perf, iter_small_boost_deque) {
     return iteration_bench<boost_deque_traits>(small_size);
 }
 
+class sum_perf {
+    struct element {
+        bi::slist_member_hook<> l_next;
+        element* sl_next;
+        unsigned value;
+        char pad[64 - sizeof(value) - sizeof(l_next) - sizeof(sl_next)];
+        element(unsigned v) noexcept : value(v) {}
+    };
+
+    static_assert(sizeof(element) == 64);
+    static_assert(sizeof(element::l_next) == sizeof(void*));
+    std::vector<element> _elements;
+
+    std::vector<element*> _array_of_pointers;
+
+    using slist = bi::slist<element,
+        bi::constant_time_size<false>,
+        bi::cache_last<true>,
+        bi::member_hook<element, bi::slist_member_hook<>, &element::l_next>>;
+    static_assert(sizeof(slist) <= 2 * sizeof(void*));
+    slist _singly_linked_list;
```

Comment:
curious if the intrusive list was what you reached for first but found the performance overhead to be too much? we had encountered this is doubly-linked intrusive list (even with all the safe/auto stuff turned off), and hand coded a simple version of effectively the same thing and got like a 10x performance reduction. not sure if slist has the same issues.